Galaxus.Functional.NUnitExtension/Is.cs

```diff
@@ -0,0 +1,9 @@
+namespace Galaxus.Functional.NUnitExtension;
+
+public class Is : NUnit.Framework.Is
+{
+    public static NoneConstraint OptionOfNone => new NoneConstraint();
+
+    public static SomeConstraint OptionOfSome => new SomeConstraint();
+    public static ResultConstraint Result => new ResultConstraint();
```

Comment:
I wonder - why is `Option` matched directly while `Result` needs a step in-between?

`Is.OptionOfSome.WithValue(1)` vs. `Is.Result.InState.Ok.WithValue(1)`? I'd go directly to `Is.Ok` or `Is.Err` - especially since I don't really get how `Is.Result.WithValue` works

Reply:
Ok, one discussion is the naming and chaining, the other is the functionality. ` WithValue(1)` would have less functionality than it has now. Sometimes one might desire to test something else then just a exact match or the OkValue is maybe an Enumeration. Then it is convenient to have all the known flexibility of nunit at hand. 

When naming the Options, I was thinking about possible naming confusions (see other comment) and went with the reviewd scheme. While when developing the Result extensions, I thought that one lazy dev might want to write a test that a `Result<int, string>` has a speciffic int value thus implicitely beeing an `Ok`-Result. On the other hand, sometimes one might just want to test on the state not on it's value. Hence the provided flexibility. 

In generall i like the Result typed extensions more. Each term in the chain has a clear responsibility.
 - `.Result` checks the actual value to be of type Result.
 - `.InState` is a understandable linkage beween the type constraint and the state constraint, allowing to reduce the possible next constraints to be either `Ok` or `Err`
 - `Ok` and `Err` checking the actual state
 - `WithValue` allowing any constraint witch is then applied to the inner value.

The option constraints on the other hand check more then one thing at once.